### DISCUSSION TITLE
Emmiting Results from Array

### DISCUSSION THREAD

**Discussion by maederan201** [pymeasure/pymeasure#483] [Q&A]
Hi everyone!

I have several procedures where I get data from an instrument in a list (potentially converted to a np.array) which I then want to emit to a result. So far, I always use a loop over the array and emit point by point, e.g.

```
for i in range(len(time)):
       data = {'Time [s]': time[i],
                   'Response [V]': response[i],
                    'Signal [V]': signal[i]}
       self.emit('results', data)
```

Is there another way of doing it? It seems a bit tedious

**Comment by CasperSchippers** [comment]
Unfortunately, at this moment there is not a much better way of doing this. You could have a look at a PR that tries to do something related: #277.

Regarding the loop, one thing you could do to (slightly) optimize it is to zip the lists you are using rather than using an index, and pass the dict directly to `emit` rather than storing it in `data`:
```
for t, r, s in zip(time, response, signal):
    self.emit('results', {'Time [s]': t, 'Response [V]': r, 'Signal [V]': s})
```

**Reply by maederan201** [reply]
This is a good suggestions, thanks!

I will have a look at #277, this could definitely be interesting.
